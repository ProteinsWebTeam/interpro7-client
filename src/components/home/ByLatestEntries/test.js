// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { ByLatestEntries } from '.';

const renderer = new ShallowRenderer();

describe('<ByEntriesFeatured />', () => {
  test('By Entries Featured', () => {
    renderer.render(
      <ByLatestEntries
        data={{
          payload: {
            results: [
              {
                extra_fields: {
                  counters: {
                    domain_architectures: 0,
                    proteins: 504,
                    proteomes: 141,
                    sets: 0,
                    structures: 0,
                    taxa: 220,
                  },
                },
                metadata: {
                  accession: 'IPR026652',
                  member_databases: {
                    panther: {
                      PTHR46657: 'FAMILY NOT NAMED',
                    },
                  },
                  name: 'Centrosomal protein of 128kDa',
                  source_database: 'interpro',
                  type: 'family',
                },
              },
            ],
          },
        }}
      />,
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
