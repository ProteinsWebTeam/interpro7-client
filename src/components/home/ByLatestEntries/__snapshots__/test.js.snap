// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ByEntriesFeatured /> By Entries Featured 1`] = `
<div
  className="feat-entry-list"
>
  <div
    className="row"
  >
    <div
      className="columns"
    >
      <Memo(ConnectFunction)
        className="card-wrapper"
        element="div"
      >
        <LatestEntry
          entry={
            Object {
              "extra_fields": Object {
                "counters": Object {
                  "domain_architectures": 0,
                  "proteins": 504,
                  "proteomes": 141,
                  "sets": 0,
                  "structures": 0,
                  "taxa": 220,
                },
              },
              "metadata": Object {
                "accession": "IPR026652",
                "member_databases": Object {
                  "panther": Object {
                    "PTHR46657": "FAMILY NOT NAMED",
                  },
                },
                "name": "Centrosomal protein of 128kDa",
                "source_database": "interpro",
                "type": "family",
              },
            }
          }
        />
      </Memo(ConnectFunction)>
      <Memo(ConnectFunction)
        className="button margin-bottom-none"
        to={
          Object {
            "description": Object {
              "entry": Object {
                "db": "InterPro",
              },
              "main": Object {
                "key": "entry",
              },
            },
            "search": Object {
              "latest_entries": "",
            },
          }
        }
      >
        View all latest entries
      </Memo(ConnectFunction)>
    </div>
  </div>
</div>
`;
